test/blackbox-tests/test-cases/pkg/default-git-branch.t

```diff
@@ -0,0 +1,40 @@
+Modify the default branch of a source repo
+
+  $ . ./helpers.sh
+
+  $ src="_git_source"
+
+  $ mkdir $src && cd $src
+  $ git init --quiet
+  $ git checkout -b "branch1"
+  Switched to a new branch 'branch1'
+  $ echo "branch 1" > file
+  $ git add -A
+  $ git commit --quiet -m "branch 1"
+  $ git checkout -b branch2
+  Switched to a new branch 'branch2'
+  $ echo "branch 2" > file
+  $ git add -A
+  $ git commit --quiet -m "branch 2"
+  $ cd ..
+
+  $ make_lockdir
+  $ make_lockpkg foo <<EOF
+  > (source (fetch (url "git+file://$PWD/$src")))
+  > (version 0.0.1)
+  > (build (run cat file))
+  > EOF
+
+Build the package
+
+  $ build_pkg foo
+  branch 2
+
+Change the default branch
+
+  $ git -C $src checkout branch1
+  Switched to branch 'branch1'
+
+And now rebuild
+
+  $ build_pkg foo
```

Comment:
Potentially worth adding a reference to https://github.com/ocaml/dune/issues/10063 so once this is fixed it is clear what the changed test output refers to.